ktor-utils/common/src/io/ktor/util/collections/ConcurrentList.kt

```diff
@@ -139,7 +139,7 @@ public class ConcurrentList<T> : MutableList<T> {
         override fun next(): T = this@ConcurrentList[current++]
 
         override fun remove() {
-            removeAt(current)
+            removeAt(current - 1)
```

Comment:
Shouldn't we fix `curent` after removal? Consider the folloing scenario (add a test like this):

```
    // given l is ConcurrentList of 1, 2, 3
    val it = l.listIterator()

    println(it.next()) // 1
    println(it.next()) // 2
    it.remove()
    println(it.next()) // crash: index out of bounds, should be 3
```

Reply:
nice catch, thanks